lib/events/stream.go

```diff
@@ -1073,6 +1085,22 @@ func (r *ProtoReader) Read(ctx context.Context) (apievents.AuditEvent, error) {
 				if !errors.Is(err, io.EOF) {
 					return nil, r.setError(trace.ConvertSystemError(err))
 				}
+
+				// due to a bug in older versions of teleport it was possible that padding
+				// bytes would end up inside of the gzip section of the archive. we should
+				// skip any dangling data in the gzip secion.
+				n, err := io.CopyBuffer(io.Discard, r.gzipReader.inner, r.messageBytes[:])
+				if err != nil {
+					return nil, r.setError(trace.ConvertSystemError(err))
+				}
+
+				if n != 0 {
+					// warn about the dangling data
+					log.WithFields(log.Fields{
+						"length": n,
+					}).Warn("skipped dangling data in session recording section")
```

Comment:
Is this warning going to be useful for cluster admins? I assumed that it'd be mostly useful for us, which would point more towards the debug level. Though perhaps it can cover other unexpected cases which would warrant the warn level? I'm not familiar enough with the architecture of session recordings to judge that.